library/ssl_tls12_server.c

```diff
@@ -2807,6 +2903,61 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
     if( ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_ECJPAKE )
     {
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        unsigned char *out_p = ssl->out_msg + ssl->out_msglen;
+        unsigned char *end_p = ssl->out_msg + MBEDTLS_SSL_OUT_CONTENT_LEN -
+                               ssl->out_msglen;
+        psa_status_t status;
+        size_t output_offset = 0;
+        size_t output_len;
+        size_t ec_len;
+
+#if !defined(MBEDTLS_ECJPAKE_ALT)
+        psa_pake_operation_t* pake_op = &(ssl->handshake->psa_pake_ctx);
+
+        mbedtls_ecp_tls_write_group( &(pake_op->ctx.ecjpake.grp),
+                                    &ec_len, out_p + output_offset,
+                                    end_p - out_p);
+#else
+        const mbedtls_ecp_curve_info *curve_info;
+
+        if( ( curve_info = mbedtls_ecp_curve_info_from_grp_id( MBEDTLS_ECP_DP_SECP256R1 ) ) == NULL )
```

Comment:
Note: we'll probably want to do this differently in the future, but that's [another discussion](https://github.com/Mbed-TLS/mbedtls/issues/6004). (Pointing it out mostly just to get a backlink in the other discussion so it's referenced there when we come to it.)

Reply:
I think that it depends on how we'll solve issue #6004. If that function is just moved out of the `ecp.c` module, then it's only a matter of renaming the function.

Reply:
Sure, I was not asking for anything now, just making a note for later.